From f3b91188d7c519e9db1c33001d4ca26d67e4a59c Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Wed, 17 Feb 2016 08:36:00 -0500
Subject: [PATCH] Fix regression due to previous bug fix

Php code to trigger this exception is

$imagick = new Imagick();
try {
    $imagick->pingImage('bug.pdf');
}
catch(ImagickException $e) {
    print $e->getMessage() . "\n";
}

Regression was introduced by https://github.com/ImageMagick/ImageMagick/commit/c4e63ad30bc42da691f2b5f82a24516dd6b4dc7

bug: https://github.com/ImageMagick/ImageMagick/issues/132
bug-debian: https://bugs.debian.org/835488
origin: upstream, https://github.com/ImageMagick/ImageMagick/commit/f6242e725c819a69bee2a444f8e4a3c7718b2b3f
---
 magick/quantum.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

Index: imagemagick-6.6.9.7/magick/quantum.c
===================================================================
--- imagemagick-6.6.9.7.orig/magick/quantum.c	2016-11-15 15:59:30.697063550 -0500
+++ imagemagick-6.6.9.7/magick/quantum.c	2016-11-15 15:59:30.693063501 -0500
@@ -303,7 +303,6 @@
   const QuantumInfo *quantum_info,const QuantumType quantum_type)
 {
   size_t
-    extent,
     packet_size;
 
   assert(quantum_info != (QuantumInfo *) NULL);
@@ -325,10 +324,9 @@
     case CbYCrYQuantum: packet_size=4; break;
     default: break;
   }
-  extent=MagickMax(image->columns,image->rows);
   if (quantum_info->pack == MagickFalse)
-    return((size_t) (packet_size*extent*((quantum_info->depth+7)/8)));
-  return((size_t) ((packet_size*extent*quantum_info->depth+7)/8));
+    return((size_t) (packet_size*image->columns*((quantum_info->depth+7)/8)));
+  return((size_t) ((packet_size*image->columns*quantum_info->depth+7)/8));
 }
 
 /*
